### ISSUE TITLE
type: the opacity of tailwind's custom keyframes cannot be a number.

### ISSUE THREAD

**Issue by HydenLiu** [tailwindlabs/tailwindcss#14247]
**What version of Tailwind CSS are you using?**

v3.4.10

**What build tool (or framework if it abstracts the build tool) are you using?**

For example: postcss-cli 8.3.1, Next.js 10.0.9, webpack 5.28.0

**What version of Node.js are you using?**

For example: v12.0.0

**What browser are you using?**

For example: Chrome, Safari, or N/A

**What operating system are you using?**

macOS

**Reproduction URL**

https://play.tailwindcss.com/ks4NiyjRok?file=config

**Describe your issue**

The opacity of tailwind's custom keyframes cannot be a number.
<img width="805" alt="image" src="https://github.com/user-attachments/assets/c10cd97c-8e88-44e3-8c14-a48397d033a0">

**Comment by wongjn** [comment]
As a work-around, you could have the values as strings and it shouldn't affect the output of the CSS: https://play.tailwindcss.com/prkqHZsQ5U?file=config

**Comment by HydenLiu** [comment]
> As a work-around, you could have the values as strings and it shouldn't affect the output of the CSS: [play.tailwindcss.com/prkqHZsQ5U?file=config](https://play.tailwindcss.com/prkqHZsQ5U?file=config)

Yeah, thanks,  I know that would fix it, but I think it's a bug and the numbers should be supported as well

**Comment by RobinMalfait** [comment]
Hey! 

This is done on purpose because we everything is stringified into a CSS string. While unlikely, this also solves an issue where some calculations could result in valid TypeScript `number` values, but that are not valid in CSS, e.g.: `NaN` or `Infinity`.

I would recommend to just use the string value, in this case of `opacity: '1'`
